src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -94,6 +103,23 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
             }
         }
 
+        private async void UpdateImdsUriApiVersionAsync(ICoreLogger logger)
+        {
+            Uri imdsErrorUri = new Uri("http://169.254.169.254/metadata/instance");
+            HttpResponse response = await _httpManager.SendGetAsync(imdsErrorUri, Headers, logger).ConfigureAwait(false);
+
+            LocalImdsErrorResponse errorResponse = JsonHelper.DeserializeFromJson<LocalImdsErrorResponse>(response.Body);
+            ImdsUri = buildImdsUri(errorResponse.NewestVersion[0]);
```

Comment:
HttpResponse can be smth else rather than OK
errorResponse can be null
errorResponse.NewestVersion can be null
errorResponse.NewestVersion[0] can be null

Reply:
Added the checks